api/src/main/java/com/capitalone/dashboard/config/LoggingFilter.java

```diff
@@ -0,0 +1,464 @@
+import javax.servlet.ServletResponse;
+import javax.servlet.WriteListener;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+public class LoggingFilter implements Filter {
+
+
+    private static final Logger LOGGER = Logger.getLogger(LoggingFilter.class);
+
+    @Autowired
+    private RequestLogRepository requestLogRepository;
+
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+        HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+
+        Map<String, String> requestMap = this.getTypesafeRequestMap(httpServletRequest);
+        BufferedRequestWrapper bufferedRequest = new BufferedRequestWrapper(httpServletRequest);
+        BufferedResponseWrapper bufferedResponse = new BufferedResponseWrapper(httpServletResponse);
+
+
+        RequestLog requestLog = new RequestLog();
+        requestLog.setClient(httpServletRequest.getRemoteAddr());
+        requestLog.setEndpoint(httpServletRequest.getRequestURI());
+        requestLog.setMethod(httpServletRequest.getMethod());
+        requestLog.setParameter(requestMap.toString());
+        requestLog.setRequestBody(JSON.parse(bufferedRequest.getRequestBody()));
+
+        chain.doFilter(bufferedRequest, bufferedResponse);
+        requestLog.setResponseBody(JSON.parse(bufferedResponse.getContent()));
+        requestLog.setResponseCode(bufferedResponse.getStatus());
+        requestLog.setTimestamp(System.currentTimeMillis());
+        final StringBuilder logMessage = new StringBuilder("REST Request - ")
+                .append("[")
+                .append(httpServletRequest.getMethod())
+                .append("] [PATH:")
+                .append(httpServletRequest.getPathInfo())
+                .append("] [PARAMETERS:")
+                .append(requestMap)
+                .append("] [BODY:")
+                .append(bufferedRequest.getRequestBody())
+                .append("] [REMOTE:")
+                .append(httpServletRequest.getRemoteAddr())
+                .append("] [STATUS:")
+                .append(bufferedResponse.getStatus())
+                .append("]");
+        LOGGER.info(logMessage.toString());
```

Comment:
don't toString unless u absolutely need to.. i dont think u want to dump the body at info, maybe only if mongo save fails..